indra/sources/dgi/processor.py

```diff
@@ -0,0 +1,243 @@
+# -*- coding: utf-8 -*-
+
+"""Process statements from the `Drug Gene Interaction Database (DGI-DB) <http://www.dgidb.org>`_."""
+
+import logging
+from typing import Iterable, List, Optional, Set, Type
+
+import pandas as pd
+
+from .api import get_version_df
+from ...ontology.standardize import get_standard_agent
+from ...statements import Activation, Complex, DecreaseAmount, Evidence, IncreaseAmount, Inhibition, Statement
+
+__all__ = [
+    'DGIProcessor',
+]
+
+logger = logging.getLogger(__name__)
+
+
+class DGIProcessor:
+    """Processor to extract INDRA Statements from DGI content.
+
+    Parameters
+    ----------
+    version : str
+        The version of DGI to use
+    """
+
+    #: A list of INDRA Statements that were extracted from DGI content.
+    statements: List[Statement]
+
+    def __init__(
+        self,
+        df: Optional[pd.DataFrame] = None,
+        version: Optional[str] = None,
+        skip_databases: Optional[Set[str]] = None,
+    ):
+        if df is None:
+            self.version, df = get_version_df(version=version)
+        else:
+            self.version = version
+        self.df = process_df(df)
+        self.statements = []
+        self.skip_databases = set(['DrugBank'] if skip_databases is None else skip_databases)
+        self.skipped = 0
+
+    def extract_statements(self) -> List[Statement]:
```

Comment:
There should be a corresponding function in the `api` file along the lines of
```
def process_df(df):
   dp = DGIProcessor(df)
   dp.extract_statements()
   return dp
```
Another endpoint in the `api` could be
```
def process_version(version=None):
    version, df  = get_version_df(version)
    return process_df(df)
````

Reply:
Done in 367d4ad3a with type annotations + docs